### DISCUSSION TITLE
What's the best way to deal with glare?

### DISCUSSION THREAD

**Discussion by smaldd14** [jomjol/AI-on-the-edge-device#2659] [Q&A]
Hey guys. I've successfully gotten my esp32 cam setup and running this repo correctly. I just wanted to reach out if you have any tips/tricks in dealing with glare?

this seems to cause the readings to be off. (see my attached images) You can see that it picks up 480 as a reading, but it should be all 0s

I've messed around with the configuration LED intensity, brightness, contrast, saturation, etc.
I've also messed around with the focus of the camera itself a bunch. This helps, but does not completely resolve the issue.

Are there any ways to completely get rid of the glare, or train the AI to disregard it?

<img width="1108" alt="Screenshot 2023-10-17 at 4 40 11 PM" src="https://github.com/jomjol/AI-on-the-edge-device/assets/18663671/ae8c6a5c-29bf-48e6-910d-26c1e37df049">
<img width="1082" alt="Screenshot 2023-10-17 at 4 40 35 PM" src="https://github.com/jomjol/AI-on-the-edge-device/assets/18663671/2b96cdcb-6121-4937-9982-9487a14e4136">

**Comment by caco3** [comment]
Search the discussions for **reflection** and **diffusion**, there where already some discussions.
You can also try to rotate the camera so the LED has a different position.

Over all, the reflection might only be a problem on the most left digit, the others should be ok. But your ROIs are not according the docs, Please follow https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/#how-to-setup-the-digit-rois-perfectly

And if you can, decrease the distance of the camera to the meter to increase the resolution.

**Comment by schneich** [comment]
This post might help you out: https://github.com/jomjol/AI-on-the-edge-device/discussions/1949
